heir_py/pipeline.py

```diff
@@ -0,0 +1,106 @@
+"""The compilation pipeline."""
+
+from pathlib import Path
+import importlib
+import sys
+import tempfile
+
+from heir_py.openfhe_config import OpenFHEConfig, DEFAULT_INSTALLED_OPENFHE_CONFIG
+from heir_py.mlir_emitter import TextualMlirEmitter
+from heir_py.pybind_helpers import pybind11_includes, pyconfig_ext_suffix
+from numba.core.bytecode import ByteCode, FunctionIdentity
+from numba.core.interpreter import Interpreter
+from heir_py.heir_backend import HeirOptBackend, HeirTranslateBackend
+from heir_py.clang import ClangBackend
+
+
+def run_compiler(
+    function,
+    openfhe_config: OpenFHEConfig = DEFAULT_INSTALLED_OPENFHE_CONFIG
+):
+    """Run the compiler."""
+    # The temporary workspace dir is so that heir-opt, heir-translate, and
+    # clang can have places to write their output files. It is cleaned up once
+    # the function returns, at which point the compiled python module has been
+    # loaded into memory and the raw files are not needed.
+    #
+    # For debugging, add delete=False to TemporaryDirectory (python3.12+)
```

Comment:
Can you expose this via an optional arg on `run_compiler` (maybe something more general such as `debug_mode`)? I'm guessing it'll be something we use a lot in the near- to mid-term